backend/tests/test_postgresql_execution_connector.py

```
@@ -0,0 +1,143 @@
+
+    assert exc_info.value.deny_code == DENY_SOURCE_BINDING_MISMATCH
+
+
+def test_default_postgresql_query_runner_requires_psycopg_driver(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    real_import = builtins.__import__
+
+    def fake_import(
+        name: str,
+        globals: dict[str, Any] | None = None,
```

Ruff (0.15.10)

[error] Function argument `globals` is shadowing a Python builtin

(A002)